src/Detector/DefaultPlatformDetector.cs

```diff
@@ -26,15 +26,18 @@ public class DefaultPlatformDetector : IDetector
             _detectorOptions = detectorOptions;
         }
 
-        public IDictionary<PlatformName, string> GetAllDetectedPlatforms(DetectorContext ctx)
+        public IDictionary<PlatformName, PlatformDetectorResult> GetAllDetectedPlatforms(DetectorContext context)
```

Comment:
PlatformDetectorResult already has Platform name as property, so how about make it return a list of `PlatformDetectorResult `

Reply:
Sure, yeah have been thinking of making that change to keep it consistent with what we already have in BuildScriptGenerator.